dragonfly/world/mcdb/provider.go

```diff
@@ -31,25 +31,27 @@ const chunkVersion = 19
 // New creates a new provider reading and writing files to files under the path passed. If a world is present
 // at the path, New will parse its data and initialise the world with it. If the data cannot be parsed, an
 // error is returned.
-func New(dir string) (*Provider, error) {
+func New(dir string) (p *Provider, previouslyExisted bool, err error) {
```

Comment:
I don't think we should add this here if it's not needed in the function. The caller can just do a os.Stat() beforehand to see if a level.dat exists.

Reply:
Removed and ran an OS stat beforehand instead